### PR TITLE
Remove long double

### PR DESCRIPTION
**Pull request by lankavitharana**

### CODE REVIEW

File: modules/ballerina-native/src/main/java/org/ballerinalang/nativeimpl/net/http/SetStatusCode.java [hunk 1]
```diff
@@ -52,8 +54,12 @@
     public BValue[] execute(Context context) {
         try {
             BMessage bMsg = (BMessage) getArgument(context, 0);
-            int statusCode = ((BInteger) getArgument(context, 1)).intValue();
-            bMsg.value().setProperty(Constants.HTTP_STATUS_CODE, statusCode);
+            long statusCode = ((BInteger) getArgument(context, 1)).intValue();
+            if (statusCode != (int) statusCode) {
+                throw BLangExceptionHelper
+                        .getRuntimeException(RuntimeErrors.INDEX_NUMBER_TOO_LARGE, statusCode); //todo, is this right?
```
Comment (SupunS) [inline modules/ballerina-native/src/main/java/org/ballerinalang/nativeimpl/net/http/SetStatusCode.java]: Remove TODOs..

File: modules/ballerina-core/src/main/java/org/ballerinalang/model/values/BArray.java [hunk 2]
```diff
@@ -51,24 +53,34 @@ public BArray(Class<V> valueClass) {
         this.valueClass = valueClass;
     }
 
-    public <V extends BValue> void add(int index, V value) {
-        ensureCapacity(index);
+    public <V extends BValue> void add(long index, V value) {
+        if (index != (int) index) {
```
Comment (lankavitharana) [inline modules/ballerina-core/src/main/java/org/ballerinalang/model/values/BArray.java]: Actually, java 8 has Math.toIntExact(long value) method to do this, it uses the same logic used here, And if we are to check index > Integer.MAX_VALUE, then we will have to do it for both sides
index < Integer.MIN_VALUE, and it is also possible.

File: modules/ballerina-native/src/main/java/org/ballerinalang/nativeimpl/connectors/data/sql/SQLDataIterator.java [hunk 3]
```diff
@@ -150,14 +118,15 @@ public double getDouble(int index) {
     }
 
     @Override
-    public double getDouble(String columnName) {
+    public double getFloat(String columnName) {
         try {
             return rs.getDouble(columnName);
```
Comment (lankavitharana) [inline modules/ballerina-native/src/main/java/org/ballerinalang/nativeimpl/connectors/data/sql/SQLDataIterator.java]: As I remember there was already a test case, will double check again.

File: modules/ballerina-native/src/main/java/org/ballerinalang/nativeimpl/net/http/SetStatusCode.java (hunk 1 above)
Comment (lankavitharana) [inline modules/ballerina-native/src/main/java/org/ballerinalang/nativeimpl/net/http/SetStatusCode.java]: need to discuss whether this is right, hence the todo. anyway is this right? even though practically status codes are integer values, is it correct to limit?

File: modules/ballerina-core/src/main/java/org/ballerinalang/natives/annotations/Utils.java [hunk 4]
```diff
@@ -17,73 +17,9 @@
...
-import org.ballerinalang.model.values.BJSON;
-import org.ballerinalang.model.values.BLong;
-import org.ballerinalang.model.values.BString;
-import org.ballerinalang.model.values.BValue;
-import org.ballerinalang.natives.exceptions.MalformedEntryException;
-
 /**
  * Common Utils methods used in Annotations.
  */
 public class Utils {
```
Comment (lankavitharana) [inline modules/ballerina-core/src/main/java/org/ballerinalang/natives/annotations/Utils.java]: done

File: modules/ballerina-native/src/main/java/org/ballerinalang/nativeimpl/connectors/data/sql/SQLDataIterator.java (hunk 3 above)
Comment (lankavitharana) [inline modules/ballerina-native/src/main/java/org/ballerinalang/nativeimpl/connectors/data/sql/SQLDataIterator.java]: Added a test case for this in SQLConnectorTest.java

File: modules/ballerina-native/src/main/java/org/ballerinalang/nativeimpl/net/http/SetStatusCode.java (hunk 1 above)
Comment (lankavitharana) [inline modules/ballerina-native/src/main/java/org/ballerinalang/nativeimpl/net/http/SetStatusCode.java]: removed the TODO, created an issue https://github.com/ballerinalang/ballerina/issues/2276 to track this

File: modules/ballerina-core/src/main/java/org/ballerinalang/model/values/BArray.java (hunk 2 above)
Comment (lankavitharana) [inline modules/ballerina-core/src/main/java/org/ballerinalang/model/values/BArray.java]: I did run a simple perf test and according to that, there is a nanosecond perf gain when using index < Integer.MAX_VALUE logic. Hence changed the logic to that

**Pull request merged**